### PR TITLE
[ICDS Dashboard] fixed the year issue on banner

### PR DESCRIPTION
**Pull request by challabeehyv**
https://dimagi-dev.atlassian.net/browse/ICDS-1157

##### SUMMARY
 fixed the year issue on banner

##### FEATURE FLAG
No

### CODE REVIEW

**Comment by challabeehyv** [comment]
Screenshot from 2020-01-08 16-32-55
Screenshot from 2020-01-08 15-50-42
Screenshot from 2020-01-08 15-31-33

**Comment by challabeehyv** [comment]
Screenshot from 2020-01-08 15-50-42
Screenshot from 2020-01-08 15-31-33

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: these variable names are quite misleading. will propose new ones. this one I think should be `yearOfPreviousMonth`

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: propose `yearOfTwoMonthsAgo`

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: propose: `yearOfNextMonth`

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: I'd recommend setting all variables to `vm.year` to start and then remove all the `else` clauses below

**Review by czue — DISMISSED** [review]
Just some stylistic changes recommended for clarity

**Review by Rohit25negi — APPROVED** [review]
Looks good. Good to go once Cory's comments are addressed

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: changed

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: changed

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: done

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: done

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: nit: I think this can just be `vm.yearOfPreviousMonth -= 1`, no?

Comment (czue) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: I think you also need to set `vm.yearOfTwoMonthsAgo` here (even if it's not used it just seems bad to leave the variable wrong).

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: oh I see how you're doing it. I guess this is ok, though is more confusing of a workflow for my brain.

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: same comment about just decrementing the value.

**Review by czue — COMMENTED** [review]
I'd still like to see this cleaned up, but I won't block it, so will leave up to @Rohit25negi / @calellowitz if they want to merge. Else if you can respond to feedback that'd be good

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: yearOfTwoMonthsAgo value will be based on yearOfPreviousMonth. So assigned it after yearOfPreviousMonth

Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: > nit: I think this can just be `vm.yearOfPreviousMonth -= 1`, no?

yearOfPreviousMonth is a string representation of the month. So cant do -=1

Comment (czue) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: Why are they strings?

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: yearOfTwoMonthsAgo value will be based on yearOfPreviousMonth. So assigned it after yearOfPreviousMonth..so had to do like this

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: they are being directly accessed in the html. so intially they might have put it like that. I just add the new variables to it

Comment (czue) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: I believe it should all work with integers and then that would simplify things substantially though, no?

Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: yes. But anyways we are not reconverting them to int rather using the current month varibale which should be fine.

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: @challabeehyv  As the Ticket looks, the priority if lower. We can use this opportunity to clean the code.
Here is some clue which you will give you some hint how you could do it here:
Screenshot 2020-01-08 at 9 40 01 PM

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: This single month subtract also moves back the Year beacuse you which you wont need to change the year manually.

Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: i was just following the earlier implementation. I will refactor it then.Thanks for  the suggest

Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: done

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: TIL: this allows negative values and works. https://www.w3schools.com/jsref/jsref_setmonth.asp

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: nit: `twoMonthsAgo` might be a clearer name and more consistent with other variables

**Review by czue — APPROVED** [review]
Thanks for hanging in there. This looks good to me.

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: changed it to twoMonthsBackDate as it seemed more appropriate

Comment (czue) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: thanks

**Pull request merged**
